.github/workflows/build-test-images.yml

```diff
@@ -0,0 +1,41 @@
+name: build-test-images
+env:
+  DOCKERHUB_USER_NAME: ${{ secrets.DOCKERHUB_USER_NAME }}
+  DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
+  MAIN_BRANCH_NAME: master
+on:
+  push:
+    branches:
+      - "**"
+
+  workflow_dispatch:
+
+jobs:
+  build_package:
+    name: build-test-images
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USER_NAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      # If current branch is not master,build and publish dev image
+      - name: Build & Push all dev images
+        if: ${{github.ref_name != env.MAIN_BRANCH_NAME }}
+        run: |
+          cd circleci/images
+          make push-all
+      # If current branch is not ,build and publish live image
+      - name: Build & Push all live images
+        if: ${{github.ref_name == env.MAIN_BRANCH_NAME }}
+        run: |
+          cd circleci/images
+          RELEASE=1 make push-all
```

Comment:
For release images I think it would make sense to _not_ parallelise the build.

The images are designed so they reuse a lot of layers. When we build them in parallel I don't think the docker cache is shared, meaning the possibility of reuse might not be understood.

Unless we can find a way that our images can reuse the common layers lets keep the release build in a single action and just wait 27 minutes.

Reply:
I agree that it makes sense to parallelize the dev images but not "release" images as with dev images, we want to see if the images are correct, and we might need to build the dev images multiple times, so speed is important there, and dev images won't be downloaded many times. However "release" images will be built once and they will be downloaded thousands of times hence layer caching will speed up things on CircleCI.

Reply:
An example can be seen in "Spin up environment" section:
https://app.circleci.com/pipelines/github/citusdata/citus/19756/workflows/d31cb87f-aa3f-40c8-8f0d-683494e2ef70/jobs/473048